sympy/solvers/inequalities.py

```diff
@@ -1004,3 +1003,241 @@ def reduce_inequalities(inequalities, symbols=[]):
+    >>> eq4 = x - z
+
+    >>> inequalities = [eq1, eq2, eq3, eq4]
+    >>> _find_pivot(inequalities)
+    y
+
+    """
+    memory={}
+    for eq in inequalities:
+        for symbol in eq.free_symbols:
+            if not(symbol in memory.keys()):
+                memory[symbol] = [False, False]
+            coeff=eq.coeff(symbol)
+            if coeff>0:
+                    memory[symbol][0] = True
+            else:
+                    memory[symbol][1] = True
+            if memory[symbol] == [True, True]:
+                return symbol
+
+
+def _split_min_max(inequalities, pivot):
+    """Split the inequalities in lists that contains the variable pivot with
+    a positive and negative sign. The list describe what pivot is
+    respectively greater and lower than according to the system of
+    inequalities. Inequalities that does not contains pivot are put in a
+    third list.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _split_min_max(
+    >>> from sympy.abc import x, y, z
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x + y + 3*z + 4
+    >>> eq4 = x - z
+
+    >>> inequalities = [eq1, eq2, eq3, eq4]
+    >>> pivot = y
+    >>> _split_min_max(inequalities, pivot)
+    ([-x - 3*z - 4], [2*x/3 + z/3 + 1/3, x + 2*z - 2], [x - z])
+
+    """
+
+    greater_than = []
+    lower_than = []
+    extra = []
+    for eq in inequalities:
+        coeff = eq.coeff(pivot)
+        if coeff > 0:
+            greater_than.append(-(eq - (pivot*coeff))/coeff)
+        elif coeff < 0:
+            lower_than.append(-(eq - (pivot*coeff))/coeff)
+        else:
+            extra.append(eq)
+    return Max(*greater_than), Min(*lower_than), extra
+
+
+def _merge_mins_maxs(maxs, mins):
+    """Build the system of inequalities which verify that all equations of maxs are greater than those of mins.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import __merge_mins_maxs
+    >>> from sympy.abc import x, z
+    >>> from sympy import Min,Max
+
+    >>> maxs = -x - 3*z - 4
+    >>> mins = Min(2*x/3 + z/3 + 1/3, x + 2*z - 2)
+
+    >>> _merge_mins_maxs(maxs, mins)
+    [2*x + 5*z + 2, 5*x/3 + 10*z/3 + 4.33333333333333]
+
+    """
+    if type(mins)!=Min:
+        mins=[mins]
+    else:
+        mins=mins.args
+
+    if type(maxs)!=Max:
+        maxs=[maxs]
+    else:
+        maxs=maxs.args
+    return [i - j for i in mins for j in maxs]
+
+
+def _fourier_motzkin(inequalities):
+    """Eliminate variables of system of linear inequalities by using Fourier-Motzkin elimination algorithm
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _fourier_motzkin
+    >>> from sympy.abc import x, y, z
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x + y + 3*z + 4
+    >>> eq4 = x - z
+
+    >>> inequalities=[eq1, eq2, eq3, eq4]
+    >>> ie, d = _fourier_motzkin(inequalities)
+    >>> ie
+    [3*x/2 + 13/10, 7*x/5 + 2/5]
```

Comment:
It would be nice to report the primitives for these since any denominator could be eliminated
```python
>>> primitive(3*x/2 + 13/10)
(1/10, 15*x + 13)
```

Reply:
You mean factorize the expression? If yes, does it improve performances? Or is this pure aesthetic?

Reply:
aesthetics... but on second thought I don't think this is necessary.